src/Xamarin.Android.Build.Tasks/Utilities/JCWGenerator.cs

```diff
@@ -245,39 +245,20 @@ static void EnsureIdenticalCollections (TaskLoggingHelper logger, NativeCodeGenS
 	{
 		logger.LogDebugMessage ($"Ensuring Java type collection in architecture '{state.TargetArch}' matches the one in architecture '{templateState.TargetArch}'");
 
-		List<TypeDefinition> templateTypes = templateState.AllJavaTypes;
-		List<TypeDefinition> types = state.AllJavaTypes;
+		var templateSet = new SortedSet<string> (templateState.AllJavaTypes.Select (t => t.FullName), StringComparer.Ordinal);
+		var typesSet = new SortedSet<string> (state.AllJavaTypes.Select (t => t.FullName), StringComparer.Ordinal);
```

Comment:
One trick, I've been doing to ensure there are no closures:
```suggestion
		var templateSet = new SortedSet<string> (templateState.AllJavaTypes.Select (static t => t.FullName), StringComparer.Ordinal);
		var typesSet = new SortedSet<string> (state.AllJavaTypes.Select (static t => t.FullName), StringComparer.Ordinal);
```
This wouldn't compile if you captured a variable on accident.

These are so short, though, it's fine as-is.